### ISSUE TITLE
maximize window for multiple users

### ISSUE THREAD

**Issue by Sertonix** [netxs-group/vtm#317]
Currently if a user maximizes a window and the another does the same a window is not maximized for the first user anymore.
This is not easy to solve since both users might have different screen sizes.

Maybe the maximized state of a window could be client side. Then the same window could have different sizes for different users.
This may be a problem for most tui applications that aren't integrated since they render depending on the screen size.

**Comment by o-sdn-o** [comment]
Yes, this is a problem. An application cannot have multiple window sizes at the same time. Each application has an internal layout of elements, which depends on the current size. For example, at one size, the position of the scrollable list will be in the lowest position, and for another size, the same list will no longer be at the bottom of the scroll.

A possible solution is to hide the maximized window for other users. I don't have a clear answer yet on how to fix this.

**Comment by o-sdn-o** [comment]
When the window is maximized by the second user, the app window should resize according to the minimum dimensions of the user's viewports.  Also forbid external observers to change app size. Allow maximize this app window only.

**Comment by o-sdn-o** [comment]
This will be a special access mode to an application instance. Let's call it Exclusive Fullscreen Application Access.

**Comment by o-sdn-o** [comment]
By maximizing an application using the menu button ` ≡ ` (Maximize/restore), the user is more likely to assume exclusive access to that application than just maximizing the window. In another way, the user can maximize the size of the window using zoom (for example, using `Ctrl+Wheel`). But they cannot get exclusive access to the application in any way.

The ` ≡ ` button of the menu should enable the full-screen exclusive mode. In this mode, a translucent stub will be placed in place of the original window, informing that this instance of the application is now in the exclusive use of some user. Exclusive access means complete isolation from other users.

**Comment by o-sdn-o** [comment]
I've made significant changes to the full-screen behavior. You can test it. Feel free to reopen this issue if you have any thoughts.
